compiler/plugins/target/AMD-AIE/iree-amd-aie/Transforms/AMDAIESplitLogicalObjectFifos.cpp

```diff
@@ -0,0 +1,214 @@
+        MemRefLayoutAttrInterface{}, oldSourceMemRefType.getMemorySpace());
+    auto newAllocOp = rewriter.create<memref::AllocOp>(rewriter.getUnknownLoc(),
+                                                       newAllocType);
+    auto newDeallocOp = rewriter.create<memref::DeallocOp>(
+        rewriter.getUnknownLoc(), newAllocOp);
+    newDeallocOp->moveBefore(&newAllocOp->getBlock()->back());
+
+    auto type = cast<MemRefType>(newAllocOp.getType());
+    // Create new logicalobjectfifo.from_memref for the newly created L2 buffer.
+    rewriter.setInsertionPoint(l2ToL1DmaOp.getSourceObjectFifo());
+    auto source = rewriter.create<AMDAIE::LogicalObjectFifoFromMemrefOp>(
+        rewriter.getUnknownLoc(), LogicalObjectFifoType::get(type),
+        newAllocOp.getResult(), sourceObjectFifo.getTiles());
+
+    SmallVector<OpFoldResult, 4> staticL3AsSourceOffsets =
+        l3ToL2DmaOp.getSourceMixedOffsets();
+    for (auto [dim, offsetToAdd] : dimToOffsetMapForL3AsSource) {
+      auto applyOp = cast<affine::AffineApplyOp>(
+          cast<Value>(staticL3AsSourceOffsets[dim]).getDefiningOp());
+      AffineMap affineMap = applyOp.getAffineMap();
+      AffineExpr affineExpr = affineMap.getResult(0);
+      AffineExpr newAffineExpr = affineExpr + offsetToAdd;
+      auto newAffineMap = AffineMap::get(/*dimCount=*/1, /*symbolCount=*/0,
+                                         {newAffineExpr}, context);
+      IRRewriter::InsertPoint oldInsertionPoint = rewriter.saveInsertionPoint();
+      rewriter.setInsertionPoint(applyOp);
+      auto newAffineApplyOp = rewriter.create<affine::AffineApplyOp>(
+          applyOp.getLoc(), newAffineMap, applyOp.getMapOperands());
+      rewriter.restoreInsertionPoint(oldInsertionPoint);
+      staticL3AsSourceOffsets[dim] = newAffineApplyOp.getResult();
+    }
+    // Create new L3 -> L2 Dma Op.
+    rewriter.setInsertionPoint(l3ToL2DmaOp);
+    rewriter.create<AMDAIE::DmaCpyNdOp>(
+        l3ToL2DmaOp.getLoc(), source, llvm::ArrayRef(staticL2AsTargetOffsets),
+        llvm::ArrayRef(staticL2AsTargetSizes),
+        llvm::ArrayRef(staticL2AsTargetStrides), l3ToL2DmaOp.getSource(),
+        llvm::ArrayRef(staticL3AsSourceOffsets),
+        llvm::ArrayRef(staticL2AsTargetSizes),
+        l3ToL2DmaOp.getSourceMixedStrides());
+
+    // Create new L2 -> L1 Input DmaOp.
+    rewriter.setInsertionPoint(l2ToL1DmaOp);
+    auto newL2ToL1DmaOp = rewriter.create<AMDAIE::DmaCpyNdOp>(
+        l2ToL1DmaOp.getLoc(), l2ToL1DmaOp.getTarget(),
+        l2ToL1DmaOp.getTargetMixedOffsets(), l2ToL1DmaOp.getTargetMixedSizes(),
+        l2ToL1DmaOp.getTargetMixedStrides(), source,
+        llvm::ArrayRef(staticL2AsSourceOffsets),
+        llvm::ArrayRef(staticL2AsSourceSizes),
+        llvm::ArrayRef(staticL2AsSourceStrides));
+    rewriter.replaceOp(l2ToL1DmaOp, newL2ToL1DmaOp);
+    // We have to discard non-zero offsets as subview has been replaced by a
+    // dedicated allocated memref.
```

Comment:
I think this is an old comment? Also, we should only erase offsets on the dimensions found above?